cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -1134,6 +1134,37 @@ ArrayKernelExec GeneratePhysicalInteger(detail::GetTypeId get_id) {
   }
 }
 
+template <template <typename...> class KernelGenerator, typename Op, typename... Args>
+ArrayKernelExec ArithmeticExecFromOp(detail::GetTypeId get_id) {
```

Comment:
I'm not sure it's a good idea to add this helper function inside the already crowded `codegen_internal.h`. Since it's arithmetic-specific, put it in `base_arithmetic_internal.h` perhaps?